src/main/scala/com/metl/model/BackendAdaptor.scala

```diff
@@ -550,3 +551,28 @@ class ResourceCachingAdaptor(sc:ServerConfiguration,cacheConfig:CacheConfig) ext
     super.isReady
   }
 }
+
+class D2LCachingAdaptor(cacheConfig:CacheConfig) {
```

Comment:
I make one DB connection per user per conversation.  It's in a foreach so ... could be concurrent?

a) yes... but if it's already 5s then diminishing returns?
b) I just mean that if I went into a conv then did something then came back, I'd expect to see changes, instead of waiting for an hour for cache to expire.  I don't mean that someone would be mashing the refresh button. 
c) the cached half will be the association between users and groups in D2L which is unlikely to change more frequently than... once per semester?  term?

But sure I guess the whole thing can be cached - I agree re re-entrancy.  If that's the case, is it worth going down the route of moving the cache outside and allowing it to be xml configured?